src/java.base/share/classes/java/net/NetworkInterface.java

```diff
@@ -35,12 +35,44 @@
 import java.util.stream.StreamSupport;
 
 /**
- * This class represents a Network Interface made up of a name,
- * and a list of IP addresses assigned to this interface.
- * It is used to identify the local interface on which a multicast group
- * is joined.
+ * This class represents a Network Interface.
+ * <p>
+ * A Network Interface is an abstraction encapsulating
+ * the characteristics of a Network Interface Controller, or
+ * Virtual Network adapter, which is a system hardware/software
+ * component connecting a computer, or host system, to a computer
+ * network. A Network Interface can be physical or virtual.
+ * A Network Interface has a name, zero or more assigned
+ * {@linkplain InetAddress IP addresses}, zero or more {@linkplain
+ * InterfaceAddress MAC Addresses}, and may have an index.
+ * The name is highly platform specific but a name such as "le0"
+ * is typical; it may not be unique. The index is a highly platform
+ * specific number that identifies the interface. The network
+ * configuration may change during the lifetime of the JVM.
+ * For example, the set of IP addresses assigned to a network
+ * interface can be transient and dynamically allocated, and may
+ * change at any time.
+ * <p>
+ * When obtaining a {@code NetworkInterface} instance, part of its
+ * configuration (such as its name and the list of assigned IP addresses),
+ * is reflective of its configuration at creation time.
+ * Obtaining an updated view of the network configuration may require
+ * looking up a network interface again in order to obtain a new instance.
+ * <p>
+ * Network interface instances are typically used to identify the local
+ * interface on which a multicast group is joined.
  *
```

Comment:
Several static methods in this class are factory methods, returning a new instance of a NetworkInterface, reflecting the configuration at the time of instantiation.  The network configuration may change at any time, for example DCHP lease renewal , or IPv6 aoutconfig. As such, these methods may need to be invoked agains in order to obtain a more up-to-date view of the network interfaces.

Reply:
Factory methods sounds good. Not sure we want to list what could cause the configuration to change: for virtual interfaces created by software components it could be just anything. I'll borrow part of your text and update.